website/docs/gitops-templates/templates.mdx

```diff
@@ -0,0 +1,198 @@
+---
+title: Using templates
+sidebar_position: 3
+hide_title: true
+---
+
+import TierLabel from "../_components/TierLabel";
+
+# Using templates <TierLabel tiers="enterprise" />
+
+## Organizing Templates
+
+Declare the type of a template by using the `weave.works/template-type` label. The value of the label is the name of the template type. The template type is used to group templates in the UI.
+
+Recommended template types:
+- `application` - for application templates
+- `cluster` - for cluster templates
+- `terraform` - for Terraform templates
+
+## Enabling/Disabling Template Components
+
+Enable or disable rendering of certain component sections in a template with the use of annotations. This can be done by using the `templates.weave.works/COMPONENT-enabled` annotation with a boolean value.
+
+Supported components:
+- `profiles`
+- `kustomizations`
+- `credentials`
+
+Example:
+```
+annotations:
+  templates.weave.works/profiles-enabled: true
+  templates.weave.works/kustomizations-enabled: true
+  templates.weave.works/credentials-enabled: true
+```
+
+## Default profile values
+
+Default and required profiles can be added via the template annotation `capi.weave.works/profile-INDEX` so that if the profiles section has been enabled on a template you can choose those profiles for the template.
+
+The annotation is added as the following:
+```
+annotations:
+    capi.weave.works/profile-0: '{"name": "NAME", "version": "VERSION", "editable": EDITABLE, "namespace": "NAMESPACE"}'
+```
+Where
+  - `name` -  is the name of the profile in the default profiles repository
+  - `version` -  (optional) will choose the default version
+  - `namespace` -  (optional) is the default target namespace for the profile
+  - `editable` -  (optional, default=false), allow the user to de-select this profile, making it a default instead of a requirement.
+
+## Rendering Templates
+
+Declare the render type indicating the templating language to be used to render the template by using the `renderType` spec.
```

Comment:
`spec.renderType` can be set to either:
- `envsubst` (default): where `${CLUSTER_NAME}` style syntax can be used.
- `templating`: go-templating style syntax `{{ .params.CLUSTER_NAME }}`. Where params are provided by the `.params` variable. You can also use template functions.